dsd/constraints.py

```diff
@@ -602,7 +623,22 @@ def satisfies_sequence_constraints(self, sequence: str) -> bool:
         """
         return all(constraint(sequence) for constraint in self.sequence_constraints)
 
-    def generate_sequence(self, rng: np.random.Generator) -> str:
+    def graph_hamming_distance(self, steps, previous_sequence):
```

Comment:
I'm a bit confused by this method. It looks like it doesn't make the whole graph, but just the edges for one node. If that's the case, its name is not very descriptive. Where is the whole graph actually created?